flash-patcher.py

```diff
@@ -220,293 +351,450 @@ def apply_patch(patch_file):
-                    perror("Invalid line numbers provided: " + split_line[-1])
-                    perror("Aborting...")
-                    exit(1)
-
-                try:
-                    for i in range(line_start, line_end + 1):
-                        del current_file[line_start - 1]
-                except IndexError:
-                    perror("")
-                    perror(patch_file + ", line " + str(current_line_no) + ": Out of range")
-                    perror("Line number " + str(line_end) + " out of range for file " + file_location)
-                    perror("Aborting...")
-                    exit(1)
-
-                write_to_file(file_location, current_file)
+            # Account for spaces in file name.
+            # Take everything except the first and last blocks.
+            # The first block is the command character.
+            # The last block is the line number(s).
+            short_name = " ".join(split_line[1:-1])
+            file_location = DECOMP_LOCATION_WITH_SCRIPTS / short_name
+
+            # Add the current script to the list of modified ones,
+            # i.e. keep this in the final output.
+            modified_scripts.add(file_location)
+
+            current_file = read_from_file(file_location, patch_file, current_line_no)
+            line_counts = split_line[-1].split("-")
+
+            if len(line_counts) != 2:
+                exception(
+                    """%s, line %d: Invalid syntax.
+                    Expected two integers, separated by a dash (-) (at %s)""",
+                    patch_file,
+                    current_line_no,
+                    line_stripped,
+                )
+                sys.exit(1)
+
+            try:
+                line_start = int(line_counts[0])
+                line_end = int(line_counts[1])
+            except ValueError:
+                exception(
+                    """%s, line %d: Invalid syntax.
+                    Invalid line numbers provided: %s
+                    Aborting...""",
+                    patch_file,
+                    current_line_no,
+                    split_line[-1],
+                )
+                sys.exit(1)
+
+            try:
+                for _i in range(line_start, line_end + 1):
+                    del current_file[line_start - 1]
+            except IndexError:
+                exception(
+                    """%s, line %d: Out of range.
+                    Line number %d out of range for file %s.
+                    Aborting...""",
+                    patch_file,
+                    current_line_no,
+                    line_end,
+                    file_location,
+                )
+                sys.exit(1)
+
+            write_to_file(file_location, current_file)
 
         # Unrecognized statement
         else:
-            perror("Unrecognized command: '" + split_line[0] + "', skipping (at " + patch_file + ", line " + str(current_line_no) + ")")
+            error(
```

Comment:
nit: error -> warn
(we're not stopping processing, so this shouldn't be considered as an error)